packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -732,7 +733,7 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
           ...acm.getRolesPerOperation('listen'),
         ]),
       ].map((r) => this.roleMap.get(r)!);
-      relatedAuthExpression = this.getVtlGenerator(ctx, def.name.value).generateAuthExpressionForRelationQuery(
+      relatedAuthExpression = this.getVtlGenerator(ctx, relatedModelName).generateAuthExpressionForRelationQuery(
```

Comment:
This is the change that supports auth across relationships amongst heterogeneous data sources